frontends/p4/strengthReduction.cpp

```diff
@@ -276,7 +276,13 @@ const IR::Node* DoStrengthReduction::postorder(IR::Slice* expr) {
         if (lo + shift_amt >= 0 && hi + shift_amt < shift_of->type->width_bits()) {
             expr->e0 = shift_of;
             expr->e1 = new IR::Constant(hi + shift_amt);
-            expr->e2 = new IR::Constant(lo + shift_amt); } }
+            expr->e2 = new IR::Constant(lo + shift_amt); }
```

Comment:
is this only done when types are unsigned?
does this work for signed values?

Reply:
maybe you can add a test for signed values too

Reply:
It does not work for signed value. How do we represent the result of strenght-reducing right shift on a signed value in P4.

for instance. 
```
(local_meta.s16 >> 1)[15:0]
```
How do we know if which of the following expression is correct if we do not know the value of the signed bit?
```
1s0 ++ local_meta.s16[15:1]
```
or 
```
1s1 ++ local_meta.s16[15:1]
```

Reply:
Yes, but you are not checking that the types are unsigned.
You may be applying this for signed values too.

Reply:
I have added a test showing strength reduction on shift does nothing to signed values.